iocore/net/ProxyProtocol.cc

```diff
@@ -302,6 +307,184 @@ proxy_protocol_v2_parse(ProxyProtocol *pp_info, const ts::TextView &msg)
   return 0;
 }
 
+size_t
+proxy_protocol_v1_build(uint8_t *buf, size_t max_buf_len, const ProxyProtocol &pp_info)
+{
```

Comment:
You should consider using `BufferWriter` here. I think it would noticeably simplify the code.

Reply:
Replaced with `ts::FixedBufferWriter`. It made code simple.